src/frontend/src/handler/create_source.rs

```diff
@@ -687,42 +727,97 @@ pub(crate) fn bind_all_columns(
 pub(crate) async fn bind_source_pk(
```

Comment:
The new validation checks in this function looks similar to `src/connector/src/parser/additional_columns.rs`. Is it necessary to check them twice?

Reply:
hmmm? I don't quite get the comment.
`src/connector/src/parser/additional_columns.rs` only holds a table of additional columns for which connector. And `handle_addition_columns` is inserting the additional columns to the table catalog.
We check the compatibility with `format` and `encode` here.

Reply:
My bad. Please ignore this.